drake/util/drakeGradientUtil.h

```diff
@@ -30,6 +31,31 @@ std::array<int, Size> intRange(int start)
   return ret;
 }
 
+namespace Drake {
+  // todo: recursive template to get arbitrary gradient order
+
+  // note: tried using template default values (e.g. Eigen::Dynamic), but they didn't seem to work on my mac clang
+  template <int num_vars> using TaylorVar = Eigen::AutoDiffScalar< Eigen::Matrix<double,num_vars,1> >;
+  template <int num_vars, int cols> using TaylorVec = Eigen::Matrix< TaylorVar<num_vars>, cols, 1>;
```

Comment:
As long as we have an architecture that will scale to the higher order case, I think it’s ok to wait on the implementation.

> On Sep 1, 2015, at 3:04 AM, Twan Koolen <notifications@github.com> wrote:
> 
> In drake/util/drakeGradientUtil.h <https://github.com/RobotLocomotion/drake/pull/1294#discussion_r38390702>:
> 
> > @@ -30,6 +31,31 @@ std::array<int, Size> intRange(int start)
> >    return ret;
> >  }
> >  
> > +namespace Drake {
> > +  // todo: recursive template to get arbitrary gradient order
> > +
> > +  // note: tried using template default values (e.g. Eigen::Dynamic), but they didn't seem to work on my mac clang
> > +  template <int num_vars> using TaylorVar = Eigen::AutoDiffScalar< Eigen::Matrix<double,num_vars,1> >;
> > +  template <int num_vars, int cols> using TaylorVec = Eigen::Matrix< TaylorVar<num_vars>, cols, 1>;
> Yeah, this would work. I don't have higher order yet. Should we worry about that when we need them?
> … <x-msg://19/#>
> —
> Reply to this email directly or view it on GitHub.
> 
>  <https://github.com/RobotLocomotion/drake/pull/1294#discussion_r38380262> <https://github.com/RobotLocomotion/drake/pull/1294/files#r38380262>
> —
> Reply to this email directly or view it on GitHub <https://github.com/RobotLocomotion/drake/pull/1294/files#r38390702>.
>